### PR TITLE
Add String matchAll

### PR DESCRIPTION
**Pull request by Elchi3**
Firefox: https://bugzilla.mozilla.org/show_bug.cgi?id=1435829
Chromiums: https://www.chromestatus.com/features/5520028858318848
Safari: https://bugs.webkit.org/show_bug.cgi?id=186694
Spec: https://github.com/tc39/proposal-string-matchall

Should pass once the other PR with the latest Chrome versions is merged.

(MDN pages do not yet exist, documenting these soon)

### CODE REVIEW

**Comment by Elchi3** [comment]
Looks like this is Nightly only for Firefox. Please hold on merging until I confirm.

File: javascript/builtins/RegExp.json [hunk 1]
```diff
@@ -1810,6 +1810,61 @@
...
+                "version_added": false
+              },
+              "nodejs": {
+                "version_added": false
+              },
+              "opera": {
+                "version_added": "60"
+              },
+              "opera_android": {
+                "version_added": true
```
Comment (ddbeck) [inline javascript/builtins/RegExp.json]: Is there a reason this one is `true` instead of 60? The Chrome status page says 60, at least

Comment (ddbeck) [reply javascript/builtins/RegExp.json]: This would seem to apply to the others Opera for Android entries in this PR too

Comment (Elchi3) [reply javascript/builtins/RegExp.json]: Yeah, Chromestatus is wrong about Opera for Android. Opera Android 49 is the latest known version and so they are far away from having 60. See https://github.com/mdn/browser-compat-data/pull/1712#issuecomment-416886047 (opera android data is wrong across our data thanks to chromestatus and other sources. I don't know how to deal with this yet).

Comment (ddbeck) [reply javascript/builtins/RegExp.json]: Ah, got it. Thanks for the update!

**Comment by ddbeck** [comment]
@Elchi3 I think rebasing this to `master` ought to fix the linter errors, now that #3288 is merged.

**Comment by Elchi3** [comment]
Done! :)

**Pull request merged**